### PR TITLE
Use updated models

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:

- Update models now that the cdn-broker and domain-broker databases have updated for the legacy-domain-certificate renewer
- Remove IAM calls since we now have all the needed certificate data
- Remove govcloud authentication since we were only using it for IAM
- Update encryption key logic since it's really used now

## Security considerations

None

### CODE REVIEW

File: migrator/models/cdn.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+def db_encryption_key():
+    return config.CDN_DATABASE_ENCRYPTION_KEY
+
+
+class CdnUserData(CdnModel):
+    """
+    CdnUserData is about the Let's Encrypt user associated with a CdnRoute.
+    We probably have no reason to ever think about this model in this project.
+    """
```
Comment (ccostino) [inline migrator/models/cdn.py:47]: 😆 

"Here's this model - pay no attention to it!"

In all seriousness, this is just for helping process and keep track of the data from the Let's Encrypt side, correct?  Just used internally here?

File: migrator/models/cdn.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    created_at = sa.Column(postgresql.TIMESTAMP)
+    updated_at = sa.Column(postgresql.TIMESTAMP)
+    deleted_at = sa.Column(postgresql.TIMESTAMP, index=True)
+    route_id = sa.Column(sa.Integer)
+    domain = sa.Column(sa.Text)
+    # cert_url is the Let's Encrypt URL for the certificate
+    cert_url = sa.Column(sa.Text)
+    # certificate is the actual body of the certificate chain
+    # this was used by the old broker, but the renewer uses fullchain_pem and leaf_pem instead
+    certificate = sa.Column(postgresql.BYTEA)
```
Comment (ccostino) [inline migrator/models/cdn.py:116]: This is kept here for backward compatibility purposes?

File: migrator/models/domain.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    created_at = sa.Column(postgresql.TIMESTAMP)
+    updated_at = sa.Column(postgresql.TIMESTAMP)
+    deleted_at = sa.Column(postgresql.TIMESTAMP, index=True)
+    route_guid = sa.Column(sa.Text)
+    domain = sa.Column(sa.Text)
+    # cert_url is the Let's Encrypt URL for the certificate
+    cert_url = sa.Column(sa.Text)
+    # certificate is the actual body of the certificate chain
+    # this was used by the old broker, but the renewer uses fullchain_pem and leaf_pem instead
+    certificate = sa.Column(postgresql.BYTEA)
```
Comment (ccostino) [inline migrator/models/domain.py:93]: Same note as above - kept for backward compatibility reasons?

File: migrator/models/common.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        if not len(users):
+            return None
+        lowest = users[0]
+        if len(lowest.routes) >= config.MAX_ROUTES_PER_USER:
+            return None
+        return lowest
+
+
+class ChallengeModel:
+    pass
```
Comment (ccostino) [inline migrator/models/common.py:60]: This isn't intended to do anything?

File: migrator/models/common.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+class CertificateModel:
+    @property
+    def needs_renewal(self):
+        now = datetime.datetime.now(datetime.timezone.utc)
+        return self.expires < now + datetime.timedelta(days=config.RENEW_BEFORE_DAYS)
+
+
+class OperationModel:
+    pass
```
Comment (ccostino) [inline migrator/models/common.py:43]: This isn't intended to do anything?

File: migrator/models/cdn.py (hunk 2 above)
Comment (bengerman13) [reply migrator/models/cdn.py:116]: basically, yes. I kept it on the legacy-domain-certificate-renewer so alembic wouldn't try to drop the columns. I left it here so it's easier to sync if/when the models change again

File: migrator/models/domain.py (hunk 3 above)
Comment (bengerman13) [reply migrator/models/domain.py:93]: same answer as above

File: migrator/models/common.py (hunk 5 above)
Comment (bengerman13) [reply migrator/models/common.py:43]: same as with ChallengeModel

**Pull request merged**